### PR TITLE
Ignore exceptions for immutable maps and lists.

### PR DESCRIPTION
**Pull request by robotdan**
Ignore exceptions for immutable maps and lists.

### CODE REVIEW

**Review by wied03 — APPROVED** [review]
I think this works well. My first instinct earlier was wondering whether this will swallow errors that it should not. I don't think it will though. Validation triggered by the action should be what's actually ensuring proper requests, not just "accidental validation" by serialization code.

File: src/main/java/org/primeframework/mvc/parameter/el/Expression.java [hunk 1]
```diff
@@ -263,7 +285,7 @@ private List<String> parse(String expression) throws ExpressionException {
     //   allows us to ignore this expression when allowUnknownParameters is true which will avoid un-necessary
     //   logging.
     for (String atom : list) {
-      if ("class".equals(atom)) {
+      if ("class" .equals(atom)) {
```
[the remark this answers is not in the document]
Comment (robotdan) [reply src/main/java/org/primeframework/mvc/parameter/el/Expression.java:288]: no.. I thought I had omitted it from the PR actually... it is either a bug in the IJ formatter - or there is a new rule in the Java formatter than I couldn't find. I can try and revert it.

**Comment by robotdan** [comment]
> I think this works well. My first instinct earlier was wondering whether this will swallow errors that it should not. I don't think it will though. Validation triggered by the action should be what's actually ensuring proper requests, not just "accidental validation" by serialization code.

Yeah, I think this is ok. But I may have @voidmain double check me.. it seems to me that at runtime, if we made a collection immutable, it was intentional. And generally speaking - in development we will be taking the exception.

**Comment by wied03** [comment]
> > I think this works well. My first instinct earlier was wondering whether this will swallow errors that it should not. I don't think it will though. Validation triggered by the action should be what's actually ensuring proper requests, not just "accidental validation" by serialization code.
> 
> Yeah, I think this is ok. But I may have @voidmain double check me.. it seems to me that at runtime, if we made a collection immutable, it was intentional. And generally speaking - in development we will be taking the exception.

Kind of a project specific decision outside the scope of PMVC per se, but would it be good for a project to make `allowUnknownParameters` a setting driven by properties files, e.g. such that in local development, a la Rails/Django/etc., you get the exceptions while running (not just in tests)?

File: src/main/java/org/primeframework/mvc/parameter/el/Expression.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (robotdan) [reply src/main/java/org/primeframework/mvc/parameter/el/Expression.java]: Fixed.

**Comment by robotdan** [comment]
Moved to https://github.com/prime-framework/prime-mvc/pull/58.

**Pull request closed without merging**